velox/common/base/SimdUtil.h

```diff
@@ -354,7 +354,12 @@ xsimd::batch<T, A> iota(const A& = {});
 template <typename T, typename A = xsimd::default_arch>
 xsimd::batch<T, A> setAll(T value, const A& = {}) {
   if constexpr (std::is_same_v<T, bool>) {
+#if defined(__aarch64__)
+    return xsimd::batch<T, A>(
+        xsimd::broadcast<bool, A>((bool)(value ? -1 : 0)));
```

Comment:
If it is hard to do I can refactor the bool registers to use `xsimd_batch<uint64>`.  It was probably a mistake to use `xsimd_batch<bool>` in the first place.

Reply:
> The rationale behind `bool` register is that we use each single bit to represent bool value instead of byte or word. So for a register 256 bits wide it contains 256 bool values. When you cast -1 to `bool`, it probably becomes 1 and the broadcast would fill the register with 0x1 on each lane. That is probably not the value we want here (we want all set bits in the register).

Yes, so, -1 when cast to bool would be 0x1 and when you broadcast bool '1', wouldn't all the bits in the vector register be set to 1? That is probably what you want when using this code (either using bool or uint64_t). Yes, please feel free to update the code as necessary. Thank you.